src/components/AppBody/AppBody.tsx

```diff
@@ -8,26 +9,56 @@ import { ContentStyled, MoviesFilter, MoviesGrid } from '../../App.styled';
 import { GENRES, INITIAL_SORTING_OPTIONS } from '../../shared/constants';
 
 const AppBody: React.FC = () => {
-  const [selectedGenre, setSelectedGenre] = useState('All');
-  const [selectedSort, setSelectedSort] = useState(INITIAL_SORTING_OPTIONS);
-  const { movies } = useMovies();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const genre = searchParams.get('genre') || 'All';
+  const sortBy = searchParams.get('sortBy') || INITIAL_SORTING_OPTIONS;
+  const search = searchParams.get('search') || undefined;
+  const [selectedGenre, setSelectedGenre] = useState(genre);
+  const [selectedSort, setSelectedSort] = useState(sortBy);
+  const params = useMemo(() => ({
+    filter: genre !== 'All' ? genre : undefined,
+    sortBy: sortBy,
+    search: search ? search : undefined,
+    searchBy: search ? 'title' : undefined,
+  }), [genre, sortBy, search]);
```

Comment:
**nitpick:** I think we can extract this logic in new hook (e.g. `useMoviesFilter`)

Reply:
done